syntax/value.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * DokuWiki Plugin struct (Syntax Component)
+ *
+ * @license GPL 2 http://www.gnu.org/licenses/gpl-2.0.html
+ * @author  Iain Hallam <iain@nineworlds.net>
+ */
+
+use dokuwiki\plugin\struct\meta\AggregationValue;
+use dokuwiki\plugin\struct\meta\ConfigParser;
+use dokuwiki\plugin\struct\meta\InlineConfigParser;
+use dokuwiki\plugin\struct\meta\SearchConfig;
+use dokuwiki\plugin\struct\meta\StructException;
+
+// must be run within Dokuwiki
+if(!defined('DOKU_INC')) die();
+
+class syntax_plugin_struct_value extends DokuWiki_Syntax_Plugin {
+
+    /**
+     * @return string Syntax mode type
+     */
+    public function getType() {
+        return 'substition';
+    }
+
+    /**
+     * @return int Sort order - Low numbers go before high numbers
+     */
+    public function getSort() {
+        /* 155 to place above Doku_Parser_Mode_hr, which would otherwise
+         * take precedence
+         * See https://www.dokuwiki.org/devel:parser:getsort_list
+         */
+        return 155;
+    }
+
+    /**
+     * Connect lookup pattern to lexer.
+     *
+     * @param string $mode Parser mode
+     */
+    public function connectTo($mode) {
+        /*
+         * Full syntax:
+         * ---- struct value ----
+         * <config>
+         * ----
+         */
+        $this->Lexer->addSpecialPattern('----+ *struct value *-+\n.*?\n----+', $mode, 'plugin_struct_value');
```

Comment:
I wonder how useful this syntax is. Would you really want to configure a complex filter for a single value? And if so, would you want to do it in this very blocky syntax?

Reply:
Removed.